File: Algorithms/209. Minimum Size Subarray Sum/solution.js
var minSubArrayLen = function (target, nums) {
  let sum = 0;
  let left = 0;
  let ans = Infinity;

  for (let right = 0; right < nums.length; right++) {
    sum += nums[right];

    while (sum >= target) {
      ans = Math.min(ans, right - left + 1);
      sum -= nums[left];
      left++;
    }
  }

  return ans === Infinity ? 0 : ans;
};

File: Algorithms/209. Minimum Size Subarray Sum/solution.cpp
class Solution
{
public:
    int minSubArrayLen(int target, vector<int> &nums)
    {
        int n = nums.size();
        int ans = INT_MAX;
        int currSum = 0;
        int j = 0;

        for (int i = 0; i < n; i++)
        {
            currSum += nums[i];

            while (currSum >= target)
            {
                ans = min(ans, i - j + 1);
                currSum -= nums[j];
                j++;
            }
        }

        return ans == INT_MAX ? 0 : ans;
    }
};
